### PR TITLE
[4.0] Say goodbye to JDatabaseDriver, JPagination, JInput, JForm

### PR DESCRIPTION
**Pull request by PhilETaylor**
Say goodbye to JDatabaseDriver, JDatabaseExceptionExecuting, JPagination, JInput, JForm

Almost done with these, the classmap aliases are now all cleaned except JObject which is coming next (its will be huge!)

Removing technical debt, the legacy prefixed classes that will be finally removed in 5.0

This PR removes all the JDatabaseDriver, JDatabaseExceptionExecuting, JPagination, JInput, JForm and more...

### CODE REVIEW

File: administrator/components/com_fields/src/Model/FieldModel.php [hunk 1]
```diff
@@ -985,14 +985,15 @@ public function validate($form, $data, $group = null)
 	/**
 	 * Method to allow derived classes to preprocess the form.
 	 *
-	 * @param   \JForm  $form   A JForm object.
+	 * @param   Form    $form   A JForm object.
```
Comment (brianteeman) [inline administrator/components/com_fields/src/Model/FieldModel.php]: A JForm object ?

Comment (brianteeman) [inline administrator/components/com_fields/src/Model/FieldModel.php]:
```suggestion
	 * @param   Form    $form   A Form object.
```

Comment (PhilETaylor) [reply administrator/components/com_fields/src/Model/FieldModel.php]: Thanks. Starting to go blind now...

File: administrator/components/com_users/src/Model/UserModel.php [hunk 2]
```diff
@@ -201,13 +201,14 @@ protected function loadFormData()
 	/**
 	 * Override Joomla\CMS\MVC\Model\AdminModel::preprocessForm to ensure the correct plugin group is loaded.
 	 *
-	 * @param   \JForm  $form   A \JForm object.
+	 * @param   Form    $form   A \JForm object.
```
Comment (brianteeman) [inline administrator/components/com_users/src/Model/UserModel.php]:
```suggestion
	 * @param   Form    $form   A Form object.
```

File: administrator/components/com_users/src/Model/GroupModel.php [hunk 3]
```diff
@@ -121,7 +121,7 @@ protected function loadFormData()
 	/**
 	 * Override preprocessForm to load the user plugin group instead of content.
 	 *
-	 * @param   \JForm  $form   A form object.
+	 * @param   Form    $form   A form object.
```
Comment (brianteeman) [inline administrator/components/com_users/src/Model/GroupModel.php]:
```suggestion
	 * @param   Form    $form   A Form object.
```

File: administrator/components/com_users/src/Model/MailModel.php [hunk 4]
```diff
@@ -75,7 +75,7 @@ protected function loadFormData()
 	/**
 	 * Method to preprocess the form
 	 *
-	 * @param   \JForm  $form   A form object.
+	 * @param   Form    $form   A form object.
```
Comment (brianteeman) [inline administrator/components/com_users/src/Model/MailModel.php]:
```suggestion
	 * @param   Form    $form   A Form object.
```

File: administrator/components/com_workflow/src/Model/StageModel.php [hunk 5]
```diff
@@ -369,7 +369,7 @@ public function publish(&$pks, $value = 1)
 	/**
 	 * Method to preprocess the form.
 	 *
-	 * @param   \JForm  $form   A \JForm object.
+	 * @param   Form    $form   A \JForm object.
```
Comment (brianteeman) [inline administrator/components/com_workflow/src/Model/StageModel.php]:
```suggestion
	 * @param   Form    $form   A Form object.
```

File: administrator/components/com_workflow/src/Model/WorkflowModel.php [hunk 6]
```diff
@@ -223,7 +223,7 @@ protected function loadFormData()
 	/**
 	 * Method to preprocess the form.
 	 *
-	 * @param   \JForm  $form   A \JForm object.
+	 * @param   Form    $form   A \JForm object.
```
Comment (brianteeman) [inline administrator/components/com_workflow/src/Model/WorkflowModel.php]:
```suggestion
	 * @param   Form    $form   Form object.
```

File: administrator/components/com_workflow/src/View/Transition/HtmlView.php [hunk 7]
```diff
@@ -35,7 +35,8 @@ class HtmlView extends BaseHtmlView
 	/**
 	 * From object to generate fields
```
Comment (brianteeman) [inline administrator/components/com_workflow/src/View/Transition/HtmlView.php]:
```suggestion
	 * Form object to generate fields
```

File: administrator/components/com_banners/src/Model/ClientModel.php [hunk 8]
```diff
@@ -85,7 +85,7 @@ protected function canEditState($record)
 	 * @param   array    $data      Data for the form.
 	 * @param   boolean  $loadData  True if the form is to load its own data (default case), false if not.
 	 *
-	 * @return  \JForm|boolean  A \JForm object on success, false on failure
+	 * @return  \Joomla\CMS\Form\Form|boolean  A \JForm object on success, false on failure
```
Comment (wilsonge) [inline administrator/components/com_banners/src/Model/ClientModel.php]:
```suggestion
	 * @return  \Joomla\CMS\Form\Form|boolean  A Form object on success, false on failure
```

File: administrator/components/com_banners/src/Model/DownloadModel.php [hunk 9]
```diff
@@ -52,7 +52,7 @@ protected function populateState()
 	 * @param   array    $data      Data for the form.
 	 * @param   boolean  $loadData  True if the form is to load its own data (default case), false if not.
 	 *
-	 * @return  \JForm|boolean  A JForm object on success, false on failure
+	 * @return  \Joomla\CMS\Form\Form|boolean  A JForm object on success, false on failure
```
Comment (wilsonge) [inline administrator/components/com_banners/src/Model/DownloadModel.php]:
```suggestion
	 * @return  \Joomla\CMS\Form\Form|boolean  A Form object on success, false on failure
```

File: administrator/components/com_categories/src/Model/CategoryModel.php [hunk 10]
```diff
@@ -397,7 +397,7 @@ public function validate($form, $data, $group = null)
 	/**
 	 * Method to preprocess the form.
 	 *
-	 * @param   \JForm  $form   A JForm object.
+	 * @param   Form    $form   A JForm object.
```
Comment (wilsonge) [inline administrator/components/com_categories/src/Model/CategoryModel.php]:
```suggestion
	 * @param   Form    $form   A Form object.
```

**Comment by PhilETaylor** [comment]
hmm looks like there is some overlap with another PR that has caused merge conflicts - I'll take a look at those

**Comment by PhilETaylor** [comment]
Merge conflicts resolved...

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**